tools/respec2html.js

```diff
@@ -1,141 +1,150 @@
+  default: true,
+  description: "Report ReSpec errors on stderr.",
+  name: "output-errors",
+  type: Boolean,
+}, {
+  alias: "w",
+  default: false,
+  description: "Report ReSpec warnings on stderr.",
+  name: "output-warnings",
+  type: Boolean,
+}];
 
-// Dealing with ReSpec source being loaded with scheme-relative link
-// i.e. <script src='//www.w3.org/Tools/respec/respec-w3c-common'>
-page.onResourceRequested = function(requestData, networkRequest) {
-  if (requestData.url === "file://www.w3.org/Tools/respec/respec-w3c-common") {
-    networkRequest.changeUrl("https://www.w3.org/Tools/respec/respec-w3c-common");
-  } else if (ignoreScripts && requestData.url.indexOf(ignoreScripts) === 0) {
-    networkRequest.abort();
+const tasks = {
+  showHelp() {
+    const getUsage = require("command-line-usage");
+    const appDetails = {
+      title: "respec2html",
+      description: "Converts a ReSpec source file to HTML and prints to std out.",
+      footer: "Project home: [underline]{https://github.com/w3c/respec}"
+    };
+    console.log(getUsage(optionDefinitions, appDetails));
+  },
+  writeTo(outPath, data){
+    const fsp = require("fs-promise");
+    const path = require("path");
+    return async.task(function*(){
+      let newFilePath = "";
+      if(path.isAbsolute(outPath)){
+        newFilePath = outPath;
+      } else {
+        newFilePath = path.resolve(process.cwd(), outPath); 
+      }
+      try {
+        yield fsp.writeFile(newFilePath, data, "utf-8");
+      } catch(err) {
+        console.error(err, err.stack);
+        process.exit(1);
+      }
+    });
   }
 };
 
-page.onConsoleMessage = function(msg) {
-  if (msg.match(/^ERROR: /)) {
-    errors.push(msg);
-  } else if (msg.match(/^WARN: /)) {
-    warnings.push(msg);
+async.task(function* run() {
+  const cli = require("command-line-args")(optionDefinitions);
+  let parsedArgs;
+  try {
+    parsedArgs = cli.parse();
+  } catch (err) {
+    console.error(err.stack);
+    tasks.showHelp();
+    process.exit(2);
   }
-};
-
-page.onError = function(msg) {
-  errors.push(msg);
-};
-
-page.open(source, function(status) {
-  if (status !== "success") {
-    console.error("Unable to access ReSpec source file.");
-    return phantom.exit(1);
+  if (parsedArgs.help || !parsedArgs.src) {
+    tasks.showHelp();
+    return;
   }
-  setTimeout(function() {
-    page.evaluateAsync(function() {
-      $.ajaxSetup({
-        timeout: 4000
-      });
-
-      function saveToPhantom() {
-        require(["core/ui", "ui/save-html"], function(ui, saver) {
-          saver.show(ui, respecConfig, document, respecEvents);
-          window.callPhantom({
-            html: saver.toString()
-          });
-        });
+  const Nightmare = require("nightmare");
+  const nightmare = new Nightmare({
+    show: false,
+    timeout: parsedArgs.timeout,
+  });
+  const URLParser = require("url");
+  const url = URLParser.parse(parsedArgs.src).href
+    .concat(`?cache-bust=${Math.random()}`);
```

Comment:
hmmm... the trouble is that when e.g. a spec imports MathJax, it doesn't want to be saved with the MathJax export, but with the MathJax script ready to run on execution.
There are probably other ways to solve this than network interception, but we need it before we can make the switch.